confirm.c

```diff
@@ -763,16 +765,18 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
-confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
-		confirm_query_cb cb)
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
 {
 	struct confirm_query *query;
-
-	query = malloc(sizeof(struct confirm_query));
+	query = malloc(sizeof(*query));
 	if(!query) logea(__FILE__, __LINE__, NULL);
 
+	if(flowid > CONFIRM_MAX_FLOWID) {
+		logd(LOG_WARN, "%s,%d: flowid > 127!\n", __FILE__, __LINE__);
```

Comment:
Understood.  I think the current interface is to blame (my fault).
The interface takes care of setting the fields to store the *flow
identifier* and the *probe identifier*.

I am not sure what the best approach is.  Do you have ideas?

One approach is to keep these in an outer module.  For example, if
I want to send 1024 probes at the same time, there is no way to
choose header field values to implement this (at least for ICMP).
We could let the library's user choose how it wants to store the
probe identifier, this way it would be natural to use the first bits
in the sequence number to store the probe identifier like you did.

The problem is that the library needs to match the response with the
probe.  If the user can set the flow identifier in any way he wants,
can the library do the matching without the user having to pass
a function to extract the flow identifier?